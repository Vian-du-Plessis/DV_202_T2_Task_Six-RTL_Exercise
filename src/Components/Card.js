import React, {useState} from "react";
import image from '../Assets/Image.jpg';
import styles from './Card.module.css';

export const Card = ( props ) => {

    const [status, setStatus] = useState("normal");

    return (
        <div className={ styles.cardContainer } 
            onMouseEnter={() => setStatus("Hovered")} 
            onMouseLeave={() => setStatus("Normal")}
        >
            <img 
                src={image} 
                alt={props.alt}
            />
            <div className={ styles.cardSubText }>
                <h1 aria-label="title">{ props.title }</h1>
                <h2 aria-label="sender">{ props.sender }</h2>
                <br />
                <h3 aria-label="message">{ props.message }</h3>
            </div>

        </div>
    );
};
