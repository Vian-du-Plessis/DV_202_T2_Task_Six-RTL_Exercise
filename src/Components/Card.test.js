import { render, screen } from "@testing-library/react"
import {Card} from "./Card";
import renderer from 'react-test-renderer';

const cardProperty = {
    title: "Pain",
    sender: "Your-Favourite-Lecturer",
    message: "I think you all would agree!",
    image: {
        url: "https://i.chzbgr.com/thumb800/16763397/h0B023285/funny-memes-programming-memes-developers-coding-memes-coder-coding-nerdy-memes-memes-java-memes",
        alt: "Noice Meme"
    }
}

describe("Test Card Component", () => {


    test("Should show title", () => {
        render( 
            <Card 
                {...cardProperty} 
            /> 
        );
        const titleElement = screen.getByLabelText(/title/i);

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Pain");

    })

    test("Should show sender", () => {
        render( 
            <Card 
                {...cardProperty} 
            /> 
        );
        const titleElement = screen.getByLabelText(/sender/i);

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Your-Favourite-Lecturer");
    })

    test("Should show image", () => {
        render(<Card {...cardProperty} />)
        const image = screen.getByRole('img');

        expect(image).toHaveAttribute('src', 'Image.jpg');

    })

    // SNAPSHOT TEST
    test("Snapshot of component", () => {
        const component = renderer.create(
            <Card/>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        renderer.act(() => {
            tree.props.onMouseEnter();
        })

        tree = component.toJSON();
        expect(tree).toMatchSnapshot()
    })


})