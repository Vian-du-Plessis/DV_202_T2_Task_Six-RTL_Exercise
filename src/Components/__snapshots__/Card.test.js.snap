// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card Component Snapshot of component 1`] = `
<div
  className="cardContainer"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    src="Image.jpg"
  />
  <div
    className="cardSubText"
  >
    <h1
      aria-label="title"
    />
    <h2
      aria-label="sender"
    />
    <br />
    <h3
      aria-label="message"
    />
  </div>
</div>
`;

exports[`Test Card Component Snapshot of component 2`] = `
<div
  className="cardContainer"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <img
    src="Image.jpg"
  />
  <div
    className="cardSubText"
  >
    <h1
      aria-label="title"
    />
    <h2
      aria-label="sender"
    />
    <br />
    <h3
      aria-label="message"
    />
  </div>
</div>
`;
